### PR TITLE
add the event object

### PR DESCRIPTION
**Pull request by fugufish**
adds the event object.

### CODE REVIEW

File: lib/polyn/event.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    ##
+    # @return [String] the data content type
+    attr_accessor :datacontenttype
+
+    ##
+    # @return [String] the data
+    attr_reader :data
+
+    def initialize(hash)
+      @specversion = hash.has_key?(:specversion) ? hash[:specversion] : "1.0"
```
Comment (reecebisel) [inline lib/polyn/event.rb:58]: Are we planning on forcing a `1.0` integration permanently? Or are do we want to force whatever the `CLOUD_EVENT_VERSION` is?

File: spec/polyn/event_spec.rb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          )
+        end.to raise_error(Polyn::Event::UnsupportedVersionError)
+      end
+    end
+  end
+
+  describe "#id" do
+    it "defaults to a uuid" do
+      expect(subject.id).to_not be_nil
+      expect(subject.id =~ /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/).to be_truthy
```
Comment (reecebisel) [inline spec/polyn/event_spec.rb:55]: This Regex needs to be put into a variable. That is a doozy dude.

File: spec/polyn/event_spec.rb [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    it "returns the passed id" do
+      expect(Polyn::Event.new(
+        type:   "test.event",
+        source: "/test/service",
+        id:     "12345678-1234-1234-1234-1234567890ab",
+        data:   {
+          foo: "bar",
+        },
+      ).id).to eq("12345678-1234-1234-1234-1234567890ab")
```
Comment (reecebisel) [inline spec/polyn/event_spec.rb:66]: Is this an intentionally brittle test? I'm fine if this was on purpose but future versions may want to change our expectation a little.

File: spec/polyn/event_spec.rb [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    end
+  end
+
+  describe "#type" do
+    it "returns the event type" do
+      expect(subject.type).to eq("test.event")
```
Comment (reecebisel) [inline spec/polyn/event_spec.rb:72]: Ditts

File: spec/polyn/event_spec.rb [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    end
+  end
+
+  describe "#source" do
+    it "returns the event source" do
+      expect(subject.source).to eq("/test/service")
```
Comment (reecebisel) [inline spec/polyn/event_spec.rb:78]: Double ditts

File: spec/polyn/event_spec.rb [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+    it "returns the passed time" do
+      expect(Polyn::Event.new(
+        type:   "test.event",
+        source: "/test/service",
+        time:   Time.new(2020, 1, 1, 12, 0, 0),
+        data:   {
+          foo: "bar",
+        },
+      ).time).to eq(Time.new(2020, 1, 1, 12, 0, 0))
```
Comment (reecebisel) [inline spec/polyn/event_spec.rb:97]: Triple Ditts

**Review by reecebisel — CHANGES_REQUESTED** [review]

File: spec/polyn/event_spec.rb (hunk 3 above)
Comment (fugufish) [reply spec/polyn/event_spec.rb:66]: how so?

**Pull request closed without merging**